File: field/contests/atcoder/abc261/a.py
l1,r1,l2,r2 = map(int,input().split())

li = [0]*101
for i in range(l1,r1+1):
    li[i] += 1
for i in range(l2,r2+1):
    li[i] += 1

ans = 0
for i in range(l2,r2+1):
    if li[i] == 2:
        ans += 1
print(max(0,ans-1))

File: field/contests/atcoder/abc261/src/bin/a.rs
#[allow(unused_imports)]
use itertools::Itertools;
#[allow(unused_imports)]
use proconio::{fastout, input,marker::{Chars, Bytes, Isize1, Usize1}};
#[allow(unused_imports)]
use std::{
    cmp::{max, min, Reverse},
    collections::{BTreeMap, BTreeSet, BinaryHeap, HashMap, HashSet, VecDeque},
    process::exit,
};

const INF: usize = 1 << 60;

#[fastout]
#[allow(non_snake_case, non_upper_case_globals, path_statements)]
fn main() {
    input! {
        L1:isize,R1:isize,L2:isize,R2:isize
    }

    println!("{}",max(0,min(R1,R2)-max(L1,L2)));
}
